modules/data.atmosphere/R/download.MERRA.R

```diff
@@ -101,69 +103,77 @@ download.MERRA <- function(outfolder, start_date, end_date,
     ))
 
     ## Create output file
-    if (file.exists(loc.file)) {
-      PEcAn.logger::logger.warn(
-        "Target file ", loc.file, " already exists.",
-        "It will be overwritten."
-      )
-    }
-    loc <- ncdf4::nc_create(loc.file, var_list)
-    on.exit(ncdf4::nc_close(loc), add = TRUE)
+    if (overwrite || !file.exists(loc.file)) {
 
-    # Populate output file
-    dates_yr <- dates[lubridate::year(dates) == year]
-    for (d in seq_along(dates_yr)) {
-      date <- dates_yr[[d]]
-      end <- d * 24
-      start <- end - 23
-      mostfile <- file.path(outfolder, sprintf("merra-most-%s.nc", as.character(date)))
-      nc <- ncdf4::nc_open(mostfile)
-      for (r in seq_len(nrow(merra_vars))) {
-        x <- ncdf4::ncvar_get(nc, merra_vars[r,][["MERRA_name"]])
-        ncdf4::ncvar_put(loc, merra_vars[r,][["CF_name"]], x,
-                         start = c(1, 1, start), count = c(1, 1, 24))
-      }
-      ncdf4::nc_close(nc)
-      presfile <- file.path(outfolder, sprintf("merra-pres-%s.nc", as.character(date)))
-      nc <- ncdf4::nc_open(presfile)
-      for (r in seq_len(nrow(merra_pres_vars))) {
-        x <- ncdf4::ncvar_get(nc, merra_pres_vars[r,][["MERRA_name"]])
-        ncdf4::ncvar_put(loc, merra_pres_vars[r,][["CF_name"]], x,
-                         start = c(1, 1, start), count = c(1, 1, 24))
+      loc <- loc.file
+      if (file.exists(loc.file)) {
+        PEcAn.logger::logger.warn(
+          "Target file ", loc.file, " already exists.",
+          "It will be overwritten."
+        )
+        loc <- ncdf4::nc_open(loc.file, write = TRUE)
+        on.exit(ncdf4::nc_close(loc), add = TRUE)
       }
-      ncdf4::nc_close(nc)
-      fluxfile <- file.path(outfolder, sprintf("merra-flux-%s.nc", as.character(date)))
-      nc <- ncdf4::nc_open(fluxfile)
-      for (r in seq_len(nrow(merra_flux_vars))) {
-        x <- ncdf4::ncvar_get(nc, merra_flux_vars[r,][["MERRA_name"]])
-        ncdf4::ncvar_put(loc, merra_flux_vars[r,][["CF_name"]], x,
-                         start = c(1, 1, start), count = c(1, 1, 24))
+      else {
+        loc <- ncdf4::nc_create(loc.file, var_list)
+        on.exit(ncdf4::nc_close(loc), add = TRUE)
       }
-      ncdf4::nc_close(nc)
-      lfofile <- file.path(outfolder, sprintf("merra-lfo-%s.nc", as.character(date)))
-      nc <- ncdf4::nc_open(lfofile)
-      for (r in seq_len(nrow(merra_lfo_vars))) {
-        x <- ncdf4::ncvar_get(nc, merra_lfo_vars[r,][["MERRA_name"]])
-        ncdf4::ncvar_put(loc, merra_lfo_vars[r,][["CF_name"]], x,
-                         start = c(1, 1, start), count = c(1, 1, 24))
+
+      # Populate output file
+      dates_yr <- dates[lubridate::year(dates) == year]
+      for (d in seq_along(dates_yr)) {
+        date <- dates_yr[[d]]
+        end <- d * 24
+        start <- end - 23
+        mostfile <- file.path(outfolder, sprintf("merra-most-%s.nc", as.character(date)))
```

Comment:
Yes actually the `get_merra_date` function internally takes care of the download and all the error prompts that might be needed in case of a failure. With that said, I agree that adding a few extra checks would just help us avoid any unexpected errors.